### PR TITLE
Merge branch-findEvents-V1.3 with Master

### LINKED ISSUE #85: Add findEvents command
As a marketer with many marketing events, I should be able to find a marketing event by name, so that I can quickly locate details of the marketing event without having to go through the entire event list.

### PR DESCRIPTION
**Pull request by lfrostbytee**
Currently, the Duke The Market application supports the Event class which is similar to the Person class.

As there may be multiple events, it is crucial to implement a similar findPersons feature like findEvents to help the user locate events more easily.

Let's,
* Add a new FindEventsCommand.java
* Add a new FindEventsCommandParser.java
* Add a new EventTitleContainsKeywordsPredicate.java
* Update Messages.java and AddressBookParser.java

Closes #85

### CODE REVIEW

File: src/main/java/seedu/address/commons/core/Messages.java [hunk 1]
```diff
@@ -8,6 +8,7 @@ public class Messages {
     public static final String MESSAGE_UNKNOWN_COMMAND = "Unknown command";
     public static final String MESSAGE_INVALID_COMMAND_FORMAT = "Invalid command format! \n%1$s";
     public static final String MESSAGE_INVALID_PERSON_DISPLAYED_INDEX = "The person index provided is invalid";
+    public static final String MESSAGE_EVENTS_LISTED_OVERVIEW = "%1$d events listed!";
     public static final String MESSAGE_PERSONS_LISTED_OVERVIEW = "%1$d persons listed!";
```
Comment (Vinodjayakumar124) [inline src/main/java/seedu/address/commons/core/Messages.java]: This change may be helpful when there's only 1 event/ 1 person listed
```suggestion
    public static final String MESSAGE_EVENTS_LISTED_OVERVIEW = "%1$d event(s) listed!";
    public static final String MESSAGE_PERSONS_LISTED_OVERVIEW = "%1$d person(s) listed!";
```

File: src/main/java/seedu/address/logic/parser/FindEventsCommandParser.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+
+import java.util.Arrays;
+
+import seedu.address.logic.commands.FindEventsCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.event.EventTitleContainsKeywordsPredicate;
+
+/**
+ * Parses input arguments and creates a new FindEventsCommand object
```
Comment (Vinodjayakumar124) [inline src/main/java/seedu/address/logic/parser/FindEventsCommandParser.java]: Just a missing fullstop
```suggestion
 * Parses input arguments and creates a new FindEventsCommand object.
```

File: src/main/java/seedu/address/logic/parser/FindEventsCommandParser.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+ */
+public class FindEventsCommandParser implements Parser<FindEventsCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the FindEventsCommand
+     * and returns a FindEventsCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
```
Comment (Vinodjayakumar124) [inline src/main/java/seedu/address/logic/parser/FindEventsCommandParser.java]: Just a missing fullstop here
```suggestion
     * @throws ParseException if the user input does not conform the expected format.
```

File: src/main/java/seedu/address/model/event/EventTitleContainsKeywordsPredicate.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+ */
+public class EventTitleContainsKeywordsPredicate implements Predicate<Event> {
+    private final List<String> keywords;
+
+    public EventTitleContainsKeywordsPredicate(List<String> keywords) {
+        this.keywords = keywords;
+    }
+
+    //TODO: Once EventTitle class is added, we need to change this to
+    //TODO: StringUtil.containsWordIgnoreCase(event.getEventTitle().fullEventTitle, kewyord));
```
Comment (Vinodjayakumar124) [inline src/main/java/seedu/address/model/event/EventTitleContainsKeywordsPredicate.java]: Using toString() might be a better option so we don't directly access the attributes of EventTitle class though I'm aware that it's a public attribute at the moment
```suggestion
    //TODO: StringUtil.containsWordIgnoreCase(event.getEventTitle().toString(), kewyord));
```

**Review by Vinodjayakumar124 — CHANGES_REQUESTED** [review]
Look good! Just some minor changes required

File: src/main/java/seedu/address/commons/core/Messages.java (hunk 1 above)
Comment (lfrostbytee) [reply src/main/java/seedu/address/commons/core/Messages.java]: Ah I like this change. Nice suggestion!

File: src/main/java/seedu/address/model/event/EventTitleContainsKeywordsPredicate.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+        this.keywords = keywords;
+    }
+
+    //TODO: Once EventTitle class is added, we need to change this to
+    //TODO: StringUtil.containsWordIgnoreCase(event.getEventTitle().toString(), kewyord));
+    @Override
+    public boolean test(Event event) {
+        return keywords.stream()
+                .anyMatch(keyword -> StringUtil.containsWordIgnoreCase(event.getEventTitle(), keyword));
+    }
```
Comment (Vinodjayakumar124) [inline src/main/java/seedu/address/model/event/EventTitleContainsKeywordsPredicate.java]: My EventTitle class will likely be merged before yours, so actually do you want to just replace this now 
```suggestion
    //TODO: StringUtil.containsWordIgnoreCase(event.getEventTitle().toString(), kewyord));
    @Override
    public boolean test(Event event) {
        return keywords.stream()
                .anyMatch(keyword -> StringUtil.containsWordIgnoreCase(event.getEventTitle().toString(), keyword));
    }
```

**Review by Vinodjayakumar124 — APPROVED** [review]
Just 1 suggestion to consider!

**Pull request merged**